sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -607,14 +632,60 @@ extension StorageBlobClient {
         manager.resumeAll()
     }
 
-    /// Retrieve a single Transfer object by its id.
+    /// Retrieve a single `Transfer` object by its id.
     ///
     /// - Parameters:
     ///   - id: The id of the transfer to retrieve.
     public func transfer(withId id: UUID) -> Transfer? {
         return manager.transfer(withId: id)
     }
 
+    /// Retrieve a single `BlobTransfer` object by name and container.
+    ///
+    /// - Parameters:
+    ///   - blob: The name of the blob.
+    ///   - container: The name of the container.
+    ///   - type: The `TransferType` to retrieve. Leave blank to retrieve either uploads or downloads.
+    /// - Returns: The matching `BlobTransfer` if found, or nil.
+    public func transfer(
+        forBlob blob: String,
+        inContainer container: String,
+        ofType type: TransferType? = nil
+    ) -> BlobTransfer? {
+        guard let encodedName = "\(container)/\(blob)".addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
+        else { return nil }
+        for transfer in manager.transfers {
+            guard let blobTransfer = transfer as? BlobTransfer else { continue }
+            guard let source = blobTransfer.source?.absoluteString else { continue }
+            guard let dest = blobTransfer.destination?.absoluteString else { continue }
+            if let transferType = type {
+                guard blobTransfer.transferType == transferType else { continue }
+                if transferType == .download, source.hasSuffix(encodedName) {
+                    return blobTransfer
+                } else if transferType == .upload, dest.hasSuffix(encodedName) {
+                    return blobTransfer
+                }
+            } else {
+                if source.hasSuffix(encodedName) || dest.hasSuffix(encodedName) {
+                    return blobTransfer
+                }
+            }
+        }
+        return nil
+    }
+
+    /// Parse a blob URL into its component parts.
+    /// - Parameter url: The complete blob URL to parse.
+    /// - Returns: A triple containing the parsed results.
+    public func parse(url: URL) -> (baseURL: String, blobName: String, containerName: String)? {
```

Comment:
This again is very helpful to have, especially give we tossed the upload/download signatures that take a URL directly and that the transfer objects store URLs exclusively.

Reply:
No longer needed.